### PR TITLE
feat: update to newer lifecycle and pack versions

### PR DESCRIPTION
**Pull request by mhdawson**
## Summary

update to newer version of lifecycle and pack

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by mhdawson** [comment]
Thanks to @natalieparellano who figured out the part about how to use the registry in the scripts.

I had to put the `docker stop` where it is as anywhere later and it did not seem to run. I suspect something to do with how the go test results are output but not sure.

**Comment by robdimsdale** [comment]
> I had to put the docker stop where it is as anywhere later and it did not seem to run. I suspect something to do with how the go test results are output but not sure.

Yeah, the `util::print::success` and `util::print::error` functions exit after printing, so that makes sense. Ideally we would refactor this to be clearer (e.g. using `trap` or similar), but it should work for now.

More generally, can you help me understand why we now need a registry? What is different from the other builder tests that run directly against the docker engine?

**Comment by natalieparellano** [comment]
> can you help me understand why we now need a registry

When extending images, we use kaniko, which requires a manifest (or more specifically, a `v1.Image` implementation) so we need the image to exist in a registry in order to pull the data.

**Comment by robdimsdale** [comment]
> When extending images, we use kaniko, which requires a manifest (or more specifically, a v1.Image implementation) so we need the image to exist in a registry in order to pull the data.

Ah. I didn't realize that. So it would be more accurate to say that we have _always_ needed a registry for this builder, we just hadn't added it yet.

File: scripts/.util/tools.json [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-  "pack": "v0.30.0-pre2"
+  "pack": "v0.30.0-rc2"
```
Comment (robdimsdale) [inline scripts/.util/tools.json:2]: Also I think pack v0.30.0 just got released so we can fix this, too.

Comment (mhdawson) [inline scripts/.util/tools.json:2]:
```suggestion
"pack": "v0.30.0
```

Comment (mhdawson) [reply scripts/.util/tools.json]: Updated, I assume it should be good, smoke tests run on PR should confirm.

**Comment by mhdawson** [comment]
@robdimsdale 
> Ah. I didn't realize that. So it would be more accurate to say that we have always needed a registry for this builder, we just hadn't added it yet.

Correct, except that because of an earlier bug in pack it at least `seemed` to run/work before.

**Comment by robdimsdale** [comment]
Ah ok. Thanks. I'm not as familiar with extensions so I'm just trying to follow along.

**Review by robdimsdale — APPROVED** [review]

**Pull request merged**